tests/integration/components/privacy/pii-test.js

```diff
@@ -0,0 +1,359 @@
+      return { id: 2, file: 1, status: 2 };
+    });
+
+    this.server.get('/v2/pii_request/:requestId/pii_data', (schema) => {
+      let pii = schema.piis.all().models;
+
+      return {
+        count: pii.length,
+        next: null,
+        previous: null,
+        results: pii.map((t) => t.attrs),
+      };
+    });
+
+    this.setProperties({
+      pii,
+      file,
+      currentOrganizationMe,
+      queryParams: {
+        app_limit: 10,
+        app_offset: 0,
+      },
+    });
+  });
+
+  test('it exists', async function (assert) {
+    await render(
+      hbs(
+        `<PrivacyModule::AppDetails::Pii @queryParams={{this.queryParams}} @file={{this.file}}/>`
+      )
+    );
+
+    assert.dom('[data-test-privacyModule-pii-piiDetectedTxt]').exists();
+
+    assert.dom('[data-test-privacyModule-pii-list]').exists();
+  });
+
+  test('it shows empty when no data', async function (assert) {
+    this.server.get('/v2/pii_request/:requestId/pii_data', () => {
+      return {
+        count: 0,
+        next: null,
+        previous: null,
+        results: [],
+      };
+    });
+
+    await render(
+      hbs(
+        `<PrivacyModule::AppDetails::Pii @queryParams={{this.queryParams}} @file={{this.file}}/>`
+      )
+    );
+
+    assert.dom('[data-test-privacyModule-pii-piiDetectedTxt]').doesNotExist();
+
+    assert.dom('[data-test-privacyModule-status]').exists();
+
+    assert
+      .dom('[data-test-privacyModule-status-header]')
+      .exists()
+      .hasText(t('privacyModule.emptyPiiHeader'));
+
+    assert
+      .dom('[data-test-privacyModule-status-desc]')
+      .exists()
+      .hasText(t('privacyModule.emptyPiiDesc'));
+  });
+
+  test('it shows correct data of pii', async function (assert) {
+    await render(
+      hbs(
+        `<PrivacyModule::AppDetails::Pii @queryParams={{this.queryParams}} @file={{this.file}}/>`
+      )
+    );
+
+    const expectedCategory = this.pii[0].type;
+
+    const allPiiCategory = findAll('[data-test-privacyModule-pii-category]');
+
+    assert.dom(allPiiCategory[0]).exists().hasText(expectedCategory);
+
+    const expectedData = this.pii[0].pii_data[0].value;
+
+    const allPiiDataValue = findAll('[data-test-privacyModule-pii-data-value]');
+
+    assert.dom(allPiiDataValue[0]).exists().hasText(expectedData);
+  });
+
+  test('it opens drawer and shows data', async function (assert) {
+    await render(
+      hbs(
+        `<PrivacyModule::AppDetails::Pii @queryParams={{this.queryParams}} @file={{this.file}}/>`
+      )
+    );
+
+    const expectedCategory = this.pii[0].type;
+
+    const expectedData = this.pii[0].pii_data[0].value;
+
+    const allPiiCategory = findAll('[data-test-privacyModule-pii-category]');
+
+    await click(allPiiCategory[0]);
+
+    const allDataFound = findAll(
+      '[data-test-privacyModule-pii-drawer-data-found]'
+    );
+
+    assert
+      .dom('[data-test-privacyModule-pii-drawer-header]')
+      .exists()
```

Comment:
We can remove `.exists()` assertions if is is chained with other assertions besides the `.dom()` assertion. This should be `assert.dom().hasText()`. This reduces the number of assertions we produce over time.